api/db/populate-attendants.js

```diff
@@ -0,0 +1,81 @@
+const { getRandomSubset, createPrompt } = require('./utils');
+
+const User = require('../models/User');
+const Event = require('../models/Event');
+const Attendant = require('../models/Attendant');
+
+let displayLogs;
+
+const createAttendant = async (attendantParams) => {
+  const attendant = new Attendant({
+    event: attendantParams.event,
+    user: attendantParams.user,
+  });
+  await attendant.save();
+
+  if (displayLogs) {
+    console.log('\n', '\x1b[0m', `New attendant created: ${attendant}`);
+  }
+};
+
+const createAttendants = async (verbose) => {
+  console.log('\n', '\x1b[0m', 'Populating attendants collection...');
+
+  let usersList = await User.distinct('_id');
+  if (usersList.length === 0) {
+    console.log(
+      '\x1b[33m',
+      'The users collection does not exist. Aborted attendants collection population.'
+    );
+    return;
+  }
+
+  let eventsList = await Event.find({
+    startDate: { $gte: new Date() },
+  });
+  if (eventsList.length === 0) {
+    console.log(
+      '\x1b[33m',
+      'The events collection does not exist. Aborted attendants collection population.'
+    );
+    return;
+  }
+
+  displayLogs = verbose;
+  let attendantsList = await Attendant.find();
+  const attendantsArray = [];
+
+  let affirmativeResponse = false;
+  if (attendantsList.length !== 0) {
+    affirmativeResponse = await createPrompt(
+      'Some events already contain attendees. Would you like to add attendees to events that already contain attendees?'
+    );
+  }
+
+  for (const event of eventsList) {
+    const attendees = attendantsList
+      .filter((attendee) => attendee.event.toString() == event._id.toString())
+      .map((attendee) => attendee.user.toString());
+
+    const availability = event.attendantsLimit - attendees.length;
+    if (!affirmativeResponse && event.attendantsLimit - 1 >= availability) {
+      continue;
+    }
+
+    const potentialAttendees = usersList.filter(
+      (user) => !attendees.includes(user.toString())
+    );
+
+    const subset = getRandomSubset(potentialAttendees, availability - 1);
+    subset.unshift(event.createdBy.toString());
```

Comment:
And same here again: Also, the `- 1` the event creator should only be added to `subset` if they are not an attendee yet.

Reply:
And same here: I left a new comment with a suggestion to try to make what I meant more clear. Let's discuss this there!